### PR TITLE
feat(lambda-nodejs): run parcel in a docker container

### PR DESCRIPTION
**Pull request by jogold**
### Commit Message
feat(lambda-nodejs): run parcel in a docker container

Redo of #7169.
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts [hunk 1]
```diff
@@ -40,74 +40,121 @@ export interface BuilderOptions {
...
+        throw new Error(`[Status ${build.status}] stdout: ${build.stdout?.toString().trim()}\n\n\nstderr: ${build.stderr?.toString().trim()}`);
       }
 
-      const args = [
-        'build', this.options.entry,
-        '--out-dir', this.options.outDir,
+      // Find the git root and mount it in the container. This allows Parcel to
+      // find the same modules/dependencies as the ones available "locally". It
+      // also supports monorepos.
+      const projectRoot = path.dirname(findGitPath());
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts:97]: @eladb this is the fix

Comment (eladb) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts:97]: This implies that the project root is the git root which means:
1. That the project directory must be inside a git repo. This is not always the case. For example, last time I checked, when you used GitHub as a source for CodePipeline, the CodeBuild project did not include a git clone but rather a copy of the source tree (without a .git directory).
2. Maybe we need to allow people to configure this?

Comment (jogold) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts:97]: Do you have maybe a suggestion to find the root of a monorepo?

Comment (eladb) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts:97]: How about we allow people to specify it and default to searching for `.git`, but if we can't find we omit an error explaining how to manually configure.

File: packages/@aws-cdk/aws-lambda-nodejs/lib/function.ts [hunk 2]
```diff
@@ -104,6 +118,7 @@ export class NodejsFunction extends lambda.Function {
       cacheDir: props.cacheDir,
       nodeVersion: extractVersion(runtime),
       nodeDockerTag: props.nodeDockerTag || `${process.versions.node}-alpine`,
+      projectRoot: path.dirname(path.resolve(projectRoot)),
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/function.ts:121]: Why dirname?

Comment (jogold) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/function.ts]: This is wrong... was because `findGitPath()` returned the path with `.git`.

**Comment by jogold** [comment]
@eladb do you have/can you share usage statistics for `aws-lambda-nodejs`?

**Pull request merged**